src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/TriviaEngine/AbstractTriviaFormatter.cs

```diff
@@ -860,8 +860,8 @@ private bool IsWhitespaceOrEndOfLine(SyntaxTrivia trivia)
             // it has line break right before it
             if (trivia2.IsElastic())
             {
-                // eat up consecutive elastic trivia or next line
-                if (trivia1.IsElastic() || IsEndOfLine(trivia1))
+                // eat up consecutive elastic trivia or first or next line
+                if (trivia1.IsElastic() || trivia1.RawKind == 0 || IsEndOfLine(trivia1))
```

Comment:
📝 Without this change, `CSharpRemoveRedundantNullableDirectiveTests.TestRedundantEnableMatchesCompilation` fails due to the introduction of a blank line at the beginning of the file. I have above average confidence in the correctness and placement of this change.

Reply:
this honestly seems odd... i don't know what all these values mean, but this is effectively saying "the first trivia is `default`", in which case, under what circumstances would the caller call into this with that value?

Reply:
>  "the first trivia is default"

It means beginning of file.

Reply:
The meaning `default` didn't hold here as expected. Am rewriting this to better detect beginning of file as a unique condition.